src/main/java/de/tum/cit/aet/artemis/core/config/websocket/WebsocketConfiguration.java

```diff
@@ -203,8 +202,7 @@ public boolean beforeHandshake(@NotNull ServerHttpRequest request, @NotNull Serv
                     @NotNull Map<String, Object> attributes) {
                 if (request instanceof ServletServerHttpRequest servletRequest) {
                     attributes.put(IP_ADDRESS, servletRequest.getRemoteAddress());
-                    Cookie jwtCookie = WebUtils.getCookie(servletRequest.getServletRequest(), JWTFilter.JWT_COOKIE_NAME);
-                    return JWTFilter.isJwtCookieValid(tokenProvider, jwtCookie);
+                    return JWTFilter.extractValidJwt((HttpServletRequest) servletRequest, tokenProvider) != null;
```

Comment:
There are failing server tests here. Is this cast really safe?

ClientForwardTest > getWebsocketEndpointWithInvalidCookie() FAILED
ClientForwardTest > getWebsocketEndpointWithCookie() FAILED
ClientForwardTest > getWebsocketEndpointFailedHandshakeNoCookie() FAILED

Caused by: java.lang.ClassCastException: class org.springframework.http.server.ServletServerHttpRequest cannot be cast to class jakarta.servlet.http.HttpServletRequest (org.springframework.http.server.ServletServerHttpRequest and jakarta.servlet.http.HttpServletRequest are in unnamed module of loader 'app')

Reply:
i will have a look, thanks